src/app/modules/group/components/user-list/user-list.component.ts

```diff
@@ -29,13 +29,16 @@ export class UserListComponent implements OnChanges {
 
           return this.getGroupMembersService.getGroupMembers(this.group.id, sort);
         } else {
-          throw new Error('group is null');
+          this.state = 'error';
```

Comment:
It is worse 😁. Actually, I missed it in the previous review, but having side effects (modifying this) in the switchMap is a very bad idea. There is a very common pattern we used often where we emit several state update in the switch map...

Reply:
Should I revert this change and keep the `throw new Error('group is null');` instead?